arithmetization/src/main/java/net/consensys/linea/extradata/LineaExtraDataPlugin.java

```diff
@@ -67,9 +67,34 @@ public void doRegister(final BesuContext context) {
   public void start() {
     super.start();
     if (profitabilityConfiguration.extraDataPricingEnabled()) {
+      final var extraDataHandler =
+          new LineaExtraDataHandler(rpcEndpointService, profitabilityConfiguration);
+      final var chainHeadHeader = blockchainService.getChainHeadHeader();
+      final var initialExtraData = chainHeadHeader.getExtraData();
+      try {
+        extraDataHandler.handle(initialExtraData);
+      } catch (final Exception e) {
+        log.warn(
+            "Failed setting initial pricing conf from extra data field ({}) of the chain head block {}({})",
+            initialExtraData,
+            chainHeadHeader.getNumber(),
+            chainHeadHeader.getBlockHash());
+      }
       besuEventsService.addBlockAddedListener(
-          new LineaExtraDataHandler(
-              rpcEndpointService, blockchainService, profitabilityConfiguration));
+          addedBlockContext -> {
+            final var importedBlockHeader = blockchainService.getChainHeadHeader();
```

Comment:
importedBlockHeader is not in the addedBlockContext event ? if yes I think it will avoid having another db access

Reply:
good catch, remove the extra query